### PR TITLE
Rework network-tools to

### PR DESCRIPTION
**Pull request by npinaeva**
1. have a single entry point
2. print help and usage examples for every function
3. be available both locally and via must-gather
4. add local-scripts folder for scripts that should be available
locally, but not vis must-gather
5. move a set of tools for networking testing to test-networking
folder
6. create scripts folder for all the other scripts
7. only included tested scripts to be available via network-tools
command, other scripts like sriov_network_info and ovn_cleanConntrack
still can be called manually via their path. That is done to make sure
support/customers don't run some dangerous scripts by accident.
They can become a part of network-tools command when we clarify usage
scenarios.

Add script template to make sure new scripts match general requirements.
Add docs-generating script that updates the second part of user.md
based on scripts help output.

Signed-off-by: Nadia Pinaeva <npinaeva@redhat.com>

### CODE REVIEW

**Comment by npinaeva** [comment]
This PR mostly needs reviewing updated docs and testing commands as explained in the docs

**Review by martinkennelly — COMMENTED** [review]
Built the fedora image and tested:
With SDN as the network plugin:
network-test: 
all good

sdn-cluster-info:
1. ~~I expected nodes folder to be populated but its empty~~ It captures one node only. I expected all nodes.
2. ~~Meta folder also isn't populated.~~
3. ~~Within cluster folder~~


sdn-node-connectivity:
~~Only log exists with error: "error: the server doesn't have a resource type "hostsubnet""~~

sdn-pod-to-pod:
all good

sdn-pod-to-svc:
all good


OVN-K incoming.

**Comment by martinkennelly** [comment]
Awesome work! This should open the floodgates hopefully ;)
With OVN-K as the network plugin:
all good

ovn-nic-firmware:
works good

ovn-pod-to-pod:
works good

ovn-pod-to-svc:
works good

File: debug-scripts/test-networking/common [hunk 1]
```diff
@@ -189,3 +192,39 @@ create_host_network_pod_on_node () {
     sleep 2
     oc wait -n "$NAMESPACE" --for=condition=Ready pod/"$POD_NAME" --timeout=3m
 }
+
+LOCAL_NS="network-tools"
+
+function ctrl_c() {
+    echo "Cleanup namespace in progress"
+    oc project default
+    oc delete ns $LOCAL_NS
```
[the remark this answers is not in the document]
Comment (npinaeva) [reply debug-scripts/test-networking/common:201]: `LOCAL_NS` is only used in `common` file, and the functions are called from other files - so I didn't want to make the calls more complex

File: debug-scripts/test-networking/ovn_pod_to_svc_connectivity [hunk 2]
```diff
@@ -48,52 +51,57 @@ do_pod_to_svc_connectivity_check () {
...
+  $USAGE
+  $USAGE <src-node-name> <dst-node-name>
+  $USAGE <src-pod-namespace>/<src-pod-name> <dst-pod-namespace>/<dst-pod-name>
+  $USAGE \"\" <dst-pod-namespace>/<dst-pod-name>
+  $USAGE <src-pod-namespace>/<src-pod-name>
+
+  oc adm must-gather --image=$NETWORK_TOOLS_IMAGE -- $USAGE
+  oc adm must-gather --image=$NETWORK_TOOLS_IMAGE -- $USAGE <src-node-name> <dst-node-name>
+  oc adm must-gather --image=$NETWORK_TOOLS_IMAGE -- $USAGE <src-pod-namespace>/<src-pod-name> <dst-svc-namespace>/<dst-svc-name>
+  oc adm must-gather --image=$NETWORK_TOOLS_IMAGE -- $USAGE \\\"\\\" <dst-svc-namespace>/<dst-svc-name>
```
[the remark this answers is not in the document]
Comment (npinaeva) [reply debug-scripts/test-networking/ovn_pod_to_svc_connectivity:89]: lol, these are examples that I expect people to run by copying from the help output, so I need it to be shown as
`oc adm must-gather --image=quay.io/openshift/origin-network-tools:latest -- network-tools sdn-pod-to-svc \"\" <dst-svc-namespace>/<dst-svc-name>`
and `\"\"` allows bash to properly parse the input as empty src and not just think `<dst-svc-namespace>/<dst-svc-name>` is the first argument

[the remark this answers is not in the document]
Comment (martinkennelly) [reply debug-scripts/test-networking/ovn_pod_to_svc_connectivity:89]: I see. Thanks.

**Comment by npinaeva** [comment]
@martinkennelly thank a lot for reviewing this!
node info really wasn't collected, should be fixed now (I also added error log to network-tools instead of just exit 1 to make such errors more visible)

**Pull request merged**